### PR TITLE
Make the start command print the endpoint on stdout

### PR DESCRIPTION
**Pull request by tobim**
###  :notebook_with_decorative_cover: Description

This is helpful for test scripts that pass in `:0` to get a random free port allocated from the operating system.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.

### CODE REVIEW

**Comment by dominiklohmann** [comment]
This needs a changelog entry, and it also needs the start command documentation to be updated accordingly.

**Review by mavam — CHANGES_REQUESTED** [review]
This feels like a not-so-nice hack. Let's try to find an alternative, or at least make it conditional on a given flag that we have to pass.

**Comment by tobim** [comment]
> This feels like a not-so-nice hack. Let's try to find an alternative, or at least make it conditional on a given flag that we have to pass.

This allow us to write:
``` bash
coproc nodefd {
  vast --verbosity=quiet -e :0 start
}
read -u "${nodefd[0]}" endpoint
# node is listening now
vast -e $endpoint status
...
vast -e $endpoint stop
```

To me this seems like very reasonable behavior and definitely not a hack. Note that reading from `stderr` instead is not that easy, because the buffer would eventually fill up and block the logging thread of the node in a `write` operation.

**Comment by mavam** [comment]
I understand where you are coming from, but I disagree that this warrants unconditional printing to stdout. Does anything speak against introducing an option that triggers this behavior, e.g., `--print-bound-port`? It's not a common case but as big consequences for launch tools that process stdout.

Also, how does your patch interact with log level "quiet"?

**Comment by dominiklohmann** [comment]
> This feels like a not-so-nice hack. Let's try to find an alternative, or at least make it conditional on a given flag that we have to pass.

To me, this feels like the proper way to solve this problem. It's the natural way to solve it for improved scriptability.

However, I also agree that this should be behind a flag, e.g., `vast.start.show-port`, and that flag should default to false.

> Also, how does your patch interact with log level "quiet"?

It shouldn't have anything to do with log levels, this is printing to stdout so it can be processed by the caller. It necessarily has to be on a separate channels from regular log output.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
-- ⚠️ `vast start` now prints the endpoint it is listening on on `stdout`.
+- ⚠️ `vast start` now prints the endpoint it is listening on on `stdout` when it
+  is invoked with the `--print-endpoint` option.
```
Comment (mavam) [inline CHANGELOG.md:16]:
```suggestion
- ⚠️ `vast start` now prints on `stdout` the endpoint it is listening on,
  provided the user added the `--print-endpoint` option.
```

File: vast.yaml.example [hunk 2]
```diff
@@ -81,6 +81,9 @@ vast:
 
   # The `vast start` command starts a new VAST server process.
   start:
+    # Prints the endpoint for clients when the server is ready to accept
+    # connections.
```
Comment (mavam) [inline vast.yaml.example:85]:
```suggestion
    # connections. This comes in handy when letting the OS choose an available
    # random port, i.e., when specifying 0 as port value.
```

**Review by mavam — APPROVED** [review]
Looks good, just a few word smithing tweaks.

File: CHANGELOG.md (hunk 1 above)
Comment (tobim) [reply CHANGELOG.md]: I took the liberty to remove "on `stdout`".

**Pull request merged**